django/db/models/fields/related_descriptors.py

```diff
@@ -193,14 +193,7 @@ def __set__(self, instance, value):
         - ``instance`` is the ``child`` instance
         - ``value`` in the ``parent`` instance on the right of the equal sign
         """
-        # If null=True, we can assign null here, but otherwise the value needs
-        # to be an instance of the related class.
-        if value is None and self.field.null is False:
```

Comment:
Thanks for the explanation. I raised it [on the mailing list](https://groups.google.com/d/topic/django-developers/4UL_FkDXX-w/discussion) and there is consensus to remove it. I think it should be done as a separate ticket for clarity (also needs to be documented). Are you able to do it? Here's the ticket: https://code.djangoproject.com/ticket/26179